src/main/java/ch/njol/skript/lang/FunctionParser.java

```diff
@@ -0,0 +1,407 @@
+		Set<FunctionReference<T>> exactReferences = new HashSet<>();
+
+		exact:
+		for (Signature<?> signature : signatures) {
+			// if arguments arent possible, skip
+			if (arguments.length > signature.getMaxParameters() || arguments.length < signature.getMinParameters()) {
+				continue;
+			}
+
+			// all remaining arguments to parse
+			// if a passed argument is named it bypasses the regular argument order of unnamed arguments
+			LinkedHashSet<String> remaining = new LinkedHashSet<>(signature.parameters().keySet());
+
+			Class<?>[] targets = new Class<?>[arguments.length];
+			for (int i = 0; i < arguments.length; i++) {
+				Argument<String> argument = arguments[i];
+
+				if (remaining.isEmpty()) {
+					continue;
+				}
+
+				Parameter<?> parameter;
+				if (argument.type() == ArgumentType.NAMED) {
+					parameter = signature.getParameter(argument.name());
+				} else {
+					parameter = signature.getParameter(remaining.getFirst());
+				}
+
+				if (parameter == null) {
+					continue exact;
+				}
+
+				if (parameter.type().isArray()) {
+					targets[i] = parameter.type().componentType();
+				} else {
+					targets[i] = parameter.type();
+				}
+
+				remaining.remove(parameter.name());
+			}
+
+			//noinspection DuplicatedCode
+			FunctionArgumentParseResult result = parseFunctionArguments(arguments, targets);
+
+			if (result.type() == FunctionArgumentParseResultType.LIST_ERROR) {
+				log.printError();
+				return null;
+			}
+
+			if (result.type() == FunctionArgumentParseResultType.OK) {
+				//noinspection unchecked
+				FunctionReference<T> reference = new FunctionReference<>(namespace, name, (Signature<T>) signature, result.parsed());
+
+				if (!reference.validate()) {
+					log.printError();
+					continue;
+				}
+
+				exactReferences.add(reference);
+			}
+		}
+		return exactReferences;
+	}
+
+	/**
+	 * Returns all possible {@link FunctionReference FunctionReferences} given the list of signatures
+	 * which only contain a single list parameter.
+	 *
+	 * @param namespace  The current namespace.
+	 * @param name       The name of the function.
+	 * @param signatures The possible signatures.
+	 * @param arguments  The passed arguments.
+	 * @param log        The logger instance.
+	 * @param <T>        The return type of the references.
+	 * @return All possible {@link FunctionReference FunctionReferences} which contain a single list parameter.
+	 */
+	private <T> Set<FunctionReference<T>> getListReferences(
+		String namespace, String name,
+		Set<Signature<?>> signatures, Argument<String>[] arguments,
+		ParseLogHandler log
+	) {
+		Set<FunctionReference<T>> references = new HashSet<>();
+
+		for (Signature<?> signature : signatures) {
+			Parameter<?> parameter = signature.parameters().firstEntry().getValue();
+
+			Class<?> target = parameter.type().componentType();
+			Class<?>[] targets = new Class<?>[]{target};
+
+			// join all args to a single arg
+			String joined = Arrays.stream(arguments).map(Argument::value)
```

Comment:
does this verify all arguments are not named/only the first is named correctly? or would it try to treat `(a: b, c: d, e)` as `(a: (b,d,e))`?